### PR TITLE
MM-15276: Migrate Team.Update to sync by default

### PR DESCRIPTION
**Pull request by therealpuneeth20**
#### Summary
This PR migrates the Update method from the TeamStore to Sync by default.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/10690

### CODE REVIEW

File: store/sqlstore/team_store.go [hunk 1]
```diff
@@ -194,41 +194,37 @@ func (s SqlTeamStore) Save(team *model.Team) store.StoreChannel {
 	})
 }
 
-func (s SqlTeamStore) Update(team *model.Team) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		team.PreUpdate()
+func (s SqlTeamStore) Update(team *model.Team) *model.AppError {
```
Comment (jespino) [inline store/sqlstore/team_store.go]: I would expect to maintain the previous behavior here, which is returning the updated team plus the error. This is consistent with other store Update calls in other stores like `User.Update` or `Channel.Update`.

File: store/storetest/team_store.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ func TestTeamStore(t *testing.T, ss store.Store) {
 	t.Run("SearchAll", func(t *testing.T) { testTeamStoreSearchAll(t, ss) })
 	t.Run("SearchOpen", func(t *testing.T) { testTeamStoreSearchOpen(t, ss) })
 	t.Run("SearchPrivate", func(t *testing.T) { testTeamStoreSearchPrivate(t, ss) })
-	t.Run("GetByIniviteId", func(t *testing.T) { testTeamStoreGetByIniviteId(t, ss) })
+	t.Run("GetByInviteId", func(t *testing.T) { testTeamStoreGetByInviteId(t, ss) })
```
Comment (jespino) [inline store/storetest/team_store.go]: Good catch! :)

File: store/storetest/team_store.go [hunk 3]
```diff
@@ -416,7 +416,9 @@ func testTeamStoreGetByIniviteId(t *testing.T, ss store.Store) {
 	}
 
 	o2.InviteId = ""
-	<-ss.Team().Update(&o2)
+	if err := ss.Team().Update(&o2); err != nil {
```
Comment (jespino) [inline store/storetest/team_store.go]: We prefer the `require.Nil(t, err)` (I know we are using them inconsistently, but we want to "migrate" to the require approach eventually).

File: store/sqlstore/team_store.go (hunk 1 above)
Comment (therealpuneeth20) [reply store/sqlstore/team_store.go]: Updated.

File: app/team.go [hunk 4]
```diff
@@ -136,12 +136,12 @@ func (a *App) UpdateTeam(team *model.Team) (*model.Team, *model.AppError) {
 }
 
 func (a *App) updateTeamUnsanitized(team *model.Team) (*model.Team, *model.AppError) {
-	result := <-a.Srv.Store.Team().Update(team)
-	if result.Err != nil {
-		return nil, result.Err
+	team, err := a.Srv.Store.Team().Update(team)
```
Comment (jespino) [inline app/team.go]: Probably here you can directly use `return a.Srv.Store.Team().Update(team)`

File: store/sqlstore/team_store.go [hunk 5]
```diff
@@ -194,41 +194,37 @@ func (s SqlTeamStore) Save(team *model.Team) store.StoreChannel {
...
-		}
+	team.PreUpdate()
 
-		oldResult, err := s.GetMaster().Get(model.Team{}, team.Id)
-		if err != nil {
-			result.Err = model.NewAppError("SqlTeamStore.Update", "store.sql_team.update.finding.app_error", nil, "id="+team.Id+", "+err.Error(), http.StatusInternalServerError)
-			return
-		}
+	if err := team.IsValid(); err != nil {
+		return team, err
```
Comment (jespino) [inline store/sqlstore/team_store.go]: I would expect to `return nil, err` only in very rare and special cases we return an object and an error, we prefer to return an object or an error.

Comment (therealpuneeth20) [reply store/sqlstore/team_store.go]: In some rare cases, when we ignore the error, the `*model.Team` object would be updated to nil.
ex:
`
team, _ := s.Update(team)
`

I can update it to `nil`, as its preferred choice.

File: app/team.go (hunk 4 above)
Comment (therealpuneeth20) [reply app/team.go]: i'm not sure, how i missed it.

File: store/sqlstore/team_store.go (hunk 5 above)
Comment (jespino) [reply store/sqlstore/team_store.go]: Yes, is the preferred choice, if I return and error, you shouldn't be using the success value at all. If you are not handling the error you must have a good reason for that. I think the unique case where we don't handle the error is when we use it for testing (and actually I think is better to handle the error there too, normally with a `require.Nil(t, err)`).

**Pull request merged**